tests/models/test_google.py

```diff
@@ -1393,3 +1394,14 @@ class CountryLanguage(BaseModel):
             ),
         ]
     )
+
+
+async def test_google_model_usage_limit_exceeded(allow_model_requests: None, google_provider: GoogleProvider):
```

Comment:
We should also add a test for VertexAI as it'll have different behavior (like including the tools etc). That won't run locally unless you have Vertex auth set up, but it'll run in CI.